tests/test_tracing.py

```diff
@@ -89,6 +99,39 @@ def test_get_func(self, frame, expected_func):
         assert get_func(frame) == expected_func
 
 
+class TestGetCallerFuncInfo:
+    @pytest.mark.parametrize(
+        "frame",
+        [
+            (GetFuncHelper.a_static_method()),
+            (GetFuncHelper.a_class_method()),
+            (GetFuncHelper().an_instance_method()),
+            (a_module_function()),
+            (GetFuncHelper().a_property),
+            (GetFuncHelper().a_cached_property),
+        ],
+    )
+    def test_get_caller_func_info(self, frame):
+        caller_frame = frame.f_back
+        if caller_frame is None:
+            return None
+        caller_code = caller_frame.f_code
+        filename = caller_code.co_filename
+        funcname = caller_code.co_name if caller_code.co_name != "<module>" else None
+        lineno = str(caller_frame.f_lineno)
```

Comment:
duplicating the code in the test doesn't test the correctness of the code, it just tests that you copy-pasted correctly :) instead please put the expected final string into the test parameters for each case above, and check against that. This may require using a placeholder for the initial part of the path which can vary depending on where the tests are run, and then replacing that placeholder inside the test with the actual path for the current test run.

Reply:
Sorry for low quality of test codes, I understand. I'll fix.